### PR TITLE
add sloppyReassign checker

### PR DESCRIPTION
**Pull request by Quasilyte**
Signed-off-by: Iskander Sharipov <quasilyte@gmail.com>

### CODE REVIEW

File: sloppyReassign_checker.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			c.warnAssignToDefine(assign, reAssigned.Name)
+			break
+		}
+	}
+}
+
+func (c *sloppyReassignChecker) warnAssignToDefine(assign *ast.AssignStmt, name string) {
+	suggest := astcopy.AssignStmt(assign)
+	suggest.Tok = token.DEFINE
+	c.ctx.Warn(assign, "replace `%s` with `%s`", assign, suggest)
```
Comment (cristaloleg) [inline sloppyReassign_checker.go]: What about something less..dominant? like `suspicious assign to the already defined variable` ?

Comment (Quasilyte) [reply sloppyReassign_checker.go]: Hah. On the one hand I agree.
On the other hand it is also a good idea to pursue short and simple-to-understand messages.
We can discuss it in our telegram chat with other maintainers.

File: testdata/sloppyReassign/negative_tests.go [hunk 2]
```diff
@@ -5,11 +5,28 @@ func ifStmtOK() (int, error) {
...
+	var errPtr *error
+	if *errPtr = returnsError(); *errPtr != nil {
+		return 0, *errPtr
+	}
+
+	// Selector expressions are fine.
+	var withError struct {
+		err error
+	}
+	if withError.err = returnsError(); withError.err != nil {
```
Comment (cristaloleg) [inline testdata/sloppyReassign/negative_tests.go]: 10/10

File: sloppyReassign_checker.go [hunk 3]
```diff
@@ -76,5 +76,5 @@ func (c *sloppyReassignChecker) VisitStmt(stmt ast.Stmt) {
 func (c *sloppyReassignChecker) warnAssignToDefine(assign *ast.AssignStmt, name string) {
 	suggest := astcopy.AssignStmt(assign)
 	suggest.Tok = token.DEFINE
-	c.ctx.Warn(assign, "replace `%s` with `%s`", assign, suggest)
+	c.ctx.Warn(assign, "re-assignment to %s can be replaced with `%s`", name, suggest)
```
Comment (cristaloleg) [inline sloppyReassign_checker.go]: re-assignment to \`%s\` can ...

 wdyt?

Comment (Quasilyte) [reply sloppyReassign_checker.go]: Done.

**Pull request merged**